packages/gatsby/src/schema/resolver-worker.js

```diff
@@ -0,0 +1,217 @@
+
+// From jest-worker library `src/types.js`
+const JEST_WORKER_CHILD_MESSAGE_IPC = 3
+
+// Contains all the types that this worker is configured to handle
+// requests for field resolvers for.
+const types = {}
+
+// List of all in-flight RPCs. keys are RPC ids, and values are
+// objects containing the `time` the RPC was sent, and the `resolve`
+// and `reject` callbacks.
+const inFlightRpcs = new Map()
+
+/**
+ * Handle rpc response by removing the original rpc from in flight
+ * rpcs and calling its `resolve` with the response
+ */
+function handleRpcResponse(rpcResponse) {
+  const { id, response } = rpcResponse
+  invariant(
+    id && response,
+    `RPC response should contain the "id" of the original RPC and a "response".`
+  )
+  const rpc = inFlightRpcs.get(id)
+  invariant(rpc, `RPC for id [${id}] not found`)
+  // Response for RPC has been received, so remove it from
+  // inFlightRpcs.
+  // TODO: Periodically expire old in flight RPCs
+  inFlightRpcs.delete(id)
+  rpc.resolve(response)
+}
+
+function handleIpc(ipc) {
+  const [rpc] = ipc
+  invariant(
+    rpc,
+    `IPC request should be an array with a single element representing the "rpc"`
+  )
+  if (rpc.type === `response`) {
+    handleRpcResponse(rpc)
+  } else {
+    throw new Error(`Invalid RPC. Must have "name" or "response"`)
+  }
+}
+
+function sendRpc({ name, args, resolve, reject }) {
+  invariant(name, `rpc name`)
+  invariant(resolve, `rpc resolve`)
+  invariant(reject, `rpc reject`)
+  // Used for randomness. There might be cheaper options
+  const id = uuidv4()
+  const rpc = { name, args, id }
+  inFlightRpcs.set(id, {
+    time: new Date(),
+    resolve,
+    reject,
+  })
+  process.send([JEST_WORKER_CHILD_MESSAGE_IPC, rpc])
+}
+
+function makeRpc(fnName) {
+  return (...args) =>
+    new Promise((resolve, reject) => {
+      sendRpc({
+        name: fnName,
+        args,
+        resolve,
+        reject,
+      })
+    })
+}
+
+// panicOnBuild will send the panic message to the parent process,
+// which will terminate, thus killing this and all other workers. All
+// other reporter functions operate locally by calling gatsby-cli
+// `reporter` directly
+function makeReporter() {
+  return Object.assign(reporter, {
+    panicOnBuild(...args) {
+      const msg = {
+        name: `reporter`,
+        args: { fnName: `panicOnBuild`, args },
+      }
+      process.send([JEST_WORKER_CHILD_MESSAGE_IPC, msg])
+    },
+  })
+}
+
+function makeRpcs(o, rpcNames) {
+  for (const rpcName of rpcNames) {
+    o[rpcName] = makeRpc(rpcName)
+  }
+  return o
+}
+
+function unsupportedFn(name) {
+  return () => {
+    throw new Error(`API [${name}] is unsupported in parallel resolver`)
+  }
+}
+
+function makeUnsupportedProps(o, props) {
+  for (const prop of props) {
+    // Defines prop so that when it's accessed for a `get`, it throws
+    // an error instead
+    Object.defineProperty(o, prop, { get: unsupportedFn(prop) })
+  }
+  return o
+}
+
+/**
```

Comment:
My [branch](https://github.com/gatsbyjs/gatsby/pull/11480) makes it so that all resolvers use stuff passed to context, instead of global ones. So here you could modify context to have the worker resolver stuff.

Reply:
@freiksenet This isn't a breaking change right? I.e will field resolvers still be able to use the closed over `getNode` etc supplied by ~setFieldsOnGraphQLNodeType~ in addition to those supplied in the context?

Reply:
I just had a read through of the branch. It looks the only api functions injected into context are `nodeModel`, which will supply `getNode` etc. Is there also a plan to include things like `getCache`, `reporter`, `createContentDigest` etc as well?